edgemesh/pkg/server/dns.go

```diff
@@ -76,20 +76,26 @@ func DnsStart() {
 
 // getDnsServer returns the specific interface ip of version 4
 func getDnsServer() (net.IP, error) {
-	ifaces, err := net.InterfaceByName(inter)
-	if err != nil {
-		return nil, err
-	}
+	for {
+		ifaces, err := net.InterfaceByName(inter)
+		if err != nil {
+			log.LOGGER.Warnf("get interface error : %s", err)
+			time.Sleep(time.Second * 5)
```

Comment:
why sleep here?

Reply:
The main reason for this bug is that containerd will only create network devices when the Pod is first created. Therefore, during the period of edge_core startup, it will always loop execution. The purpose of adding sleep is to make the cpu not too high. When you deploy the first Pod, this function will detect the corresponding device and exit normally.

Reply:
can we reduce the sleep time ? 5 seconds looks high.

Reply:
I can reduce this time to 3 seconds.